convert2rhel/subscription.py

```diff
@@ -277,28 +377,173 @@ def args(self):
+        address = register_server.Start(
+            i18n.SUBSCRIPTION_MANAGER_LOCALE, dbus_interface="com.redhat.RHSM1.RegisterServer"
+        )
+
+        try:
+            # Use the private bus to register the machine
+            loggerinst.debug("Connecting to the private DBus")
+            private_bus = dbus.connection.Connection(address)
+            loggerinst.debug("Getting a subscription-manager Register object from dbus")
+            register_object = private_bus.get_object("com.redhat.RHSM1", "/com/redhat/RHSM1/Register")
+
+            try:
+                if self.password:
+                    loggerinst.info("Registering via username/password: %s" % " ".join(hide_secrets(self.args)))
+                    register_object.Register(
+                        self.org or "",
+                        self.username,
+                        self.password,
+                        REGISTER_OPTS_DICT,
+                        self.connection_opts,
+                        i18n.SUBSCRIPTION_MANAGER_LOCALE,
+                        dbus_interface="com.redhat.RHSM1.Register",
+                    )
+
+                else:
+                    loggerinst.info("Registering via org/activation_key: %s" % " ".join(hide_secrets(self.args)))
+                    register_object.RegisterWithActivationKeys(
+                        self.org,
+                        [self.activation_key],
+                        REGISTER_OPTS_DICT,
+                        self.connection_opts,
+                        i18n.SUBSCRIPTION_MANAGER_LOCALE,
+                        dbus_interface="com.redhat.RHSM1.Register",
+                    )
+
+            except dbus.exceptions.DBusException as e:
+                # Sometimes we get NoReply but the registration has succeeded.
+                # Check the registration status before deciding if this is an error.
+                if e.get_dbus_name() == "org.freedesktop.DBus.Error.NoReply":
+                    # We need to set the connection opts in config before
+                    # checking for registration otherwise we might ask the
+                    # wrong server if the host is registered.
+                    self._set_connection_opts_in_config()
+
+                    if not _is_registered():
+                        # Host is not registered so re-raise the error
+                        raise
+                else:
+                    raise
+                # Host was registered so continue
+            else:
+                # On success, we need to set the connection opts as well
+                self._set_connection_opts_in_config()
+
+        finally:
+            # Always shut down the private bus
+            loggerinst.debug("Shutting down private DBus instance")
+            register_server.Stop(i18n.SUBSCRIPTION_MANAGER_LOCALE, dbus_interface="com.redhat.RHSM1.RegisterServer")
+
+        loggerinst.info("Successfully registered")
```

Comment:
I think you added this later after you added `loggerinst.info("System registration succeeded.")`.

Now two similar messages are printed:
```
Successfully registered
System registration succeeded.
```

Reply:
I agree.  I think I added this as I was writing unit_tests and realized that there isn't much to go on for a successful exit.  I'll remove it and depending on returning None as meaning success here.